eth/gasestimator/gasestimator.go

```diff
@@ -94,13 +96,13 @@ func Estimate(ctx context.Context, call *core.Message, opts *Options, gasCap uin
 
 		// If the allowance is larger than maximum uint64, skip checking
 		if allowance.IsUint64() && hi > allowance.Uint64() {
-			transfer := call.Value
-			if transfer == nil {
-				transfer = new(big.Int)
-			}
 			log.Debug("Gas estimation capped by limited funds", "original", hi, "balance", balance,
 				"sent", transfer, "maxFeePerGas", feeCap, "fundable", allowance)
+
 			hi = allowance.Uint64()
+			if hi < params.TxGas {
+				return 0, nil, core.ErrInsufficientFunds
+			}
```

Comment:
Sure, it's _possible_ to special-case this case. But why though -- there's no reason to believe it's a common case, and since it'll be caught by the current code, why make it special?